web/packages/teleport/src/Console/DocumentSsh/useFileTransfer.ts

```diff
@@ -0,0 +1,240 @@
+          EventType.FILE_TRANSFER_REQUEST_DENY,
+          handleFileTransferDenied
+        );
+      }
+    };
+  }, [tty]);
+
+  function removeFileTransferRequest(requestId: string) {
+    setFileTransferRequests(prevstate =>
+      prevstate.filter(ft => ft.requestID !== requestId)
+    );
+  }
+
+  async function download(
+    location: string,
+    abortController: AbortController,
+    moderatedSessionParams?: ModeratedSessionParams
+  ) {
+    const url = await getScpUrl({
+      location,
+      clusterId: currentDoc.clusterId,
+      serverId: currentDoc.serverId,
+      login: currentDoc.login,
+      filename: location,
+      moderatedSessonId: moderatedSessionParams?.moderatedSessionId,
+      fileTransferRequestId: moderatedSessionParams?.fileRequestId,
+    });
+    if (!url) {
+      // if we return nothing here, the file transfer will not be added to the
+      // file transfer list. If we add it to the list, the file will continue to
+      // start the download and return another here. This prevents a second network
+      // request that we know will fail.
+      return;
+    }
+    return getHttpFileTransferHandlers().download(url, abortController);
+  }
+
+  async function handleDownload(
+    location: string,
+    abortController: AbortController
+  ) {
+    if (session.moderated) {
+      tty.sendFileDownloadRequest(location);
+      return;
+    }
+
+    return download(location, abortController);
+  }
+
+  async function upload(
+    location: string,
+    file: File,
+    abortController: AbortController,
+    moderatedSessionParams?: ModeratedSessionParams
+  ) {
+    const url = await getScpUrl({
+      location,
+      clusterId: currentDoc.clusterId,
+      serverId: currentDoc.serverId,
+      login: currentDoc.login,
+      filename: file.name,
+      moderatedSessonId: moderatedSessionParams?.moderatedSessionId,
+      fileTransferRequestId: moderatedSessionParams?.fileRequestId,
+    });
+    if (!url) {
+      // if we return nothing here, the file transfer will not be added to the
+      // file transfer list. If we add it to the list, the file will continue to
+      // start the download and return another here. This prevents a second network
+      // request that we know will fail.
+      return;
+    }
+    return getHttpFileTransferHandlers().upload(url, file, abortController);
+  }
+
+  async function handleUpload(
+    location: string,
+    file: File,
+    abortController: AbortController
+  ) {
+    if (session.moderated) {
+      tty.sendFileUploadRequest(location, file);
+      return;
+    }
+
+    return upload(location, file, abortController);
+  }
+
+  function handleFileTransferDenied(request: FileTransferRequest) {
+    removeFileTransferRequest(request.requestID);
+  }
+
+  function handleFileTransferApproval(
+    request: FileTransferRequest,
+    file?: File
+  ) {
+    removeFileTransferRequest(request.requestID);
+    if (request.requester !== user.username) {
```

Comment:
Sure, sounds good.

Reply:
I went with exporting the function but there is another pattern throughout the codebase (for web mostly) that is like `makeUiFileTransferRequest` where other things are added, and basically morphs a response from an API into something usable by the api. I seen it a lot in our tables.

Not sure if that would be better here right now since we're only adding this one field, but if it starts to get more complicated, I'll abstract it to that in the future. 

Again, for now, exporting a function is just fine.